src/coreclr/tools/Common/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -231,16 +233,23 @@ private void EmitDelegateCall(DelegateMarshallingMethodThunk delegateMethod, PIn
 
         private void EmitPInvokeCall(PInvokeILCodeStreams ilCodeStreams)
         {
-            if (!_flags.PreserveSig && _targetMethod.Signature.ReturnType != _targetMethod.Context.GetWellKnownType(WellKnownType.Void))
-                throw new NotSupportedException();
+            if (!_flags.PreserveSig)
+            {
+                TypeDesc managedReturnType = _targetMethod.Signature.ReturnType;
+                bool supportedType = managedReturnType.IsPrimitive || managedReturnType.IsInterface || managedReturnType.IsObject;
```

Comment:
Is this going to break `void` return value type?

Reply:
No, IsPrimitive handle that because it just look at Enum value here
https://github.com/dotnet/runtime/blob/0fcd7690a14335a8807144d0e7ad937aa83699b0/src/coreclr/tools/Common/TypeSystem/Common/TypeDesc.cs#L198
and 
https://github.com/dotnet/runtime/blob/0fcd7690a14335a8807144d0e7ad937aa83699b0/src/coreclr/tools/Common/TypeSystem/Common/TypeFlags.cs#L31